extras/bindings/fortran/src/preciceFortran.cpp

```diff
@@ -55,6 +58,40 @@ try {
   std::abort();
 }
 
+void precicef_create_with_communicator_(
+    const char *participantName,
+    const char *configFileName,
+    const int  *solverProcessIndex,
+    const int  *solverProcessSize,
+    const int  *communicator,
+    int         participantNameLength,
+    int         configFileNameLength)
+try {
+#ifndef PRECICE_NO_MPI
+  // MPI_Comm_f2c requires an MPI_Fint argument, which may differ from int.
+  // However, the Fortran standard guarantees interoperability of the int type,
+  // so that is what is passed. Do the conversion in case int is not identical
+  // to MPI_Fint.
+  MPI_Fint f_communicator = (MPI_Fint) *communicator;
```

Comment:
Let's use a static cast here.
Also, we still need to check if the communicator is not a nullptr with `PRECICE_CHECK(communicator, "")`.

```suggestion
  MPI_Fint f_communicator = static_cast<MPI_Fint>(*communicator);
```